### ISSUE TITLE
`azurerm_mssql_server_transparent_data_encryption: support for Key Vault Keys from a Managed HSM

### ISSUE THREAD

**Issue by baaym** [hashicorp/terraform-provider-azurerm#22977]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.5

### AzureRM Provider Version

3.69.0

### Affected Resource(s)/Data Source(s)

azurerm_mssql_server_transparent_data_encryption

### Terraform Configuration Files

```hcl
resource "azurerm_mssql_server" "sql-server" {
  name                          = "sql-server"
  resource_group_name           = "resource-group"
  location                      = "location"
  version                       = "12.0"
  connection_policy             = "Proxy"
  administrator_login           = "a_username"
  administrator_login_password  = "a_password"

  identity {
    type = "SystemAssigned"
  }
}

# Depends on an existing HSM
resource "azurerm_mssql_server_transparent_data_encryption" "sql-server-key" {
  server_id        = azurerm_mssql_server.sql-server.id
  key_vault_key_id = "https://XXX.managedhsm.azure.net/keys/KEY_NAME/KEY_VERSION"
}
```

### Debug Output/Panic Output

```shell
Error: parsing "https://XXX.managedhsm.azure.net/keys/KEY_NAME/KEY_VERSION": internal-error: Managed HSM IDs are not supported as Key Vault Nested Items
```

### Expected Behaviour

With AzureRM version 3.38.0 this works fine, and our SQL Server uses a HSM key for TDE. It should still be possible to set a HSM key as the TDE key on Azure MSSql.

### Actual Behaviour

Terraform throws an error:
```
Error: parsing "https://XXX.managedhsm.azure.net/keys/KEY_NAME/KEY_VERSION": internal-error: Managed HSM IDs are not supported as Key Vault Nested Items
```

### Steps to Reproduce

1. `terraform plan`

### References

I traced back the change to Pull Request #22162, where `nested_item.go` now explicitly checks for the presence of `.managedhsm.` in the key ID. When the related parse function is called from `mssql_server_transparent_data_encryption_resource.go`, it trips over the presence of `.managedhsm.` and throws the error mentioned above.

Direct link to the diff: https://github.com/hashicorp/terraform-provider-azurerm/pull/22162/files#diff-47ee3cc88d70fdbce65e1d87a4f46810d37aab7ed4a292f2224273c9cd7b6d8d

**Comment by gautetk** [comment]
FYI:
This issue also seems to occur when trying to use managed HSM customer managed key on a storage account or for an encryption scope.

**Comment by katbyte** [comment]
fixed by #26496
